### PR TITLE
Support shader-printf and remove write-only texture WAR

### PR DESCRIPTION
**Pull request by kaizhangNV**
Add shader-printf support in our playground, now user can play with c style printf.

Since slang already support write-only texture type, we can now remove the WAR that using one-channel write-read texture as the storage texture and encode all color channels into one. Now we can use 4 channel texture in our playground.

TODO:
There are still two remaining issues on slang side.

1. slang doesn't support brga8 format texture, [format('bgra8')] WTexture2D ...;      will cause error.  https://github.com/shader-slang/slang/issues/5469
2. a bug in supporting bit_cast<uint64_t>(double): https://github.com/shader-slang/slang/issues/5470

### CODE REVIEW

**Review by csyonghe — APPROVED** [review]

File: compute.js [hunk 1]
```diff
@@ -1,4 +1,6 @@
 
+var sprintf = null;
```
Comment (csyonghe) [inline compute.js:2]: is this used?

Comment (kaizhangNV) [reply compute.js]: removed.

File: playgroundShader.js [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+    uint32_t low = 0;
+    uint32_t high = 0;
+};
+
+internal RWStructuredBuffer<FormatedStruct> g_printedBuffer;
+
+interface IPrintf
+{
+    uint32_t typeFlag();
+    uint32_t writePrintfWords();
```
[the remark this answers is not in the document]
Comment (kaizhangNV) [reply playgroundShader.js:24]: yea, good point, but now we can not add g_printBuffer into the function parameter. Looks like wgsl doesn't support it and we need some legalization path like glsl to make this happen.

File: try-slang.js [hunk 3]
```diff
@@ -232,35 +232,33 @@ async function printResult()
...
-    await computePipeline.outputBufferRead.mapAsync(GPUMapMode.READ);
+    await Promise.all([
+        computePipeline.outputBufferRead.mapAsync(GPUMapMode.READ),
+        computePipeline.printfBufferRead.mapAsync(GPUMapMode.READ)]);
 
     const output = new Int32Array(computePipeline.outputBufferRead.getMappedRange());
 
-    const textResult = output.toString() + "\n";
+    // debug testing:
+    let textResult = "Data Output:\n" + output.toString() + "\n\n";
```
[the remark this answers is not in the document]
Comment (kaizhangNV) [reply try-slang.js]: done

**Pull request merged**